File: IGListTest/IGListTest/Views/WeatherDetailCell.m
//
//  WeatherDetailCell.m
//  IGListTest
//
//  Created by liuruixuan on 2023/4/16.
//

#import "WeatherDetailCell.h"
#import "Utils.h"

@interface WeatherDetailCell ()



@end

@implementation WeatherDetailCell

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self.contentView addSubview:self.titleLabel];
        [self.contentView addSubview:self.detailLabel];
        self.contentView.backgroundColor = ssRGBHex(0x0c1f3f);
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
    CGRect insetBounds = UIEdgeInsetsInsetRect(self.bounds, CommonInsets);
    self.titleLabel.frame = insetBounds;
    self.detailLabel.frame = insetBounds;
}

- (UILabel *)titleLabel {
    if (!_titleLabel) {
        _titleLabel = [[UILabel alloc] init];
        _titleLabel.font = AppFontDefault;
        _titleLabel.textColor = ssRGBHex(0x42c84b);
        _titleLabel.backgroundColor = [UIColor clearColor];
    }
    return _titleLabel;
}

- (UILabel *)detailLabel {
    if (!_detailLabel) {
        _detailLabel = [[UILabel alloc] init];
        _detailLabel.font = AppFontDefault;
        _detailLabel.textColor = ssRGBHex(0x42c84b);
        _detailLabel.backgroundColor = [UIColor clearColor];
        _detailLabel.textAlignment = NSTextAlignmentRight;
    }
    return _detailLabel;
}

@end

File: Marslink_Swift5/Marslink_Swift5/Views/WeatherDetailCell.swift
//
//  WeatherDetailCell.swift
//  Marslink_Swift5
//
//  Created by liuruixuan on 2023/4/17.
//

import UIKit

class WeatherDetailCell: UICollectionViewCell {
    let titleLabel: UILabel = {
        let label = UILabel()
        label.backgroundColor = UIColor.clear
        label.font = AppFont()
        label.textColor = UIColor(hex6: 0x42C84B)
        return label
    }()

    let detailLabel: UILabel = {
        let label = UILabel()
        label.backgroundColor = UIColor.clear
        label.font = AppFont()
        label.textColor = UIColor(hex6: 0x42C84B)
        label.textAlignment = .right
        return label
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        contentView.addSubview(titleLabel)
        contentView.addSubview(detailLabel)
        contentView.backgroundColor = UIColor(hex6: 0x0C1F3F)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func layoutSubviews() {
        super.layoutSubviews()
        let insetBounds = bounds.inset(by: CommonInsets)
        titleLabel.frame = insetBounds
        detailLabel.frame = insetBounds
    }
}
